csharp/ql/src/Security Features/CWE-022/ZipSlipBroken.cs

```diff
@@ -0,0 +1,7 @@
+public static void WriteToDirectory(IArchiveEntry entry,
+                                    string destDirectory,
+                                    ExtractionOptions options){
+  string file = Path.GetFileName(entry.Key);
+  string destFileName = Path.Combine(destDirectory, file);
+  entry.WriteToFile(destFileName, options);
+}
```

Comment:
We have started formatting all C# code using 4 spaces indentation, and `{` on it's own line (following the standard that Microsoft uses). Moreover, I would like for all help examples to be compilable, so we can have a copy of the examples in the test directory as well.

The "Bad" example should always have a "class Bad" top-level (e.g. https://github.com/Semmle/ql/blob/master/csharp/ql/src/Bad%20Practices/Declarations/LocalScopeVariableShadowsMemberBad.cs), and the "Good" example should have a "class Good" top-level (e.g. https://github.com/Semmle/ql/blob/master/csharp/ql/src/Bad%20Practices/Declarations/LocalScopeVariableShadowsMemberGood.cs). The examples are then duplicated inside https://github.com/Semmle/ql/tree/master/csharp/ql/test/query-tests/Bad%20Practices/Declarations/LocalScopeVariableShadowsMember, and in case extractor options are required, I put them in a separate file https://github.com/Semmle/ql/blob/master/csharp/ql/test/query-tests/Bad%20Practices/Declarations/LocalScopeVariableShadowsMember/ExtractorOptions.cs.

Reply:
Good idea - done.